src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends { [key: string]: number | string }>(
+  items: T[],
+  key: string,
+): GroupsMap<T> {
+  return items.reduce((result: GroupsMap<T>, item: T) => {
+    if (!result[item[key]]) {
+      result[item[key]] = [];
+    }
+
+    result[item[key]].push(item);
+
+    return result;
```

Comment:
The current implementation does not handle the case where `key` does not exist on `item` or when `item[key]` is not a string or number (due to the generic constraint `T extends { [key: string]: number | string }`). This could lead to unexpected behavior. It's important to add a check to ensure that `item` has the property `key` and that the value of `item[key]` is of the expected type.